server/src/main/java/keywhiz/service/resources/BatchSecretDeliveryResource.java

```diff
@@ -0,0 +1,156 @@
+import keywhiz.service.daos.ClientDAO.ClientDAOFactory;
+import keywhiz.service.daos.SecretController;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.*;
+import java.util.Iterator;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+/**
+ * parentEndpointName secret
+ * <p>
+ * resourceDescription Retrieve Batched Named Secrets
+ */
+@Path("/batchsecret")
+@Produces(APPLICATION_JSON)
+public class BatchSecretDeliveryResource {
+  private static final Logger logger = LoggerFactory.getLogger(BatchSecretDeliveryResource.class);
+
+  private final SecretController secretController;
+  private final AclDAO aclDAO;
+  private final ClientDAO clientDAO;
+
+  @Inject
+  public BatchSecretDeliveryResource(@Readonly SecretController secretController,
+                                     AclDAOFactory aclDAOFactory, ClientDAOFactory clientDAOFactory) {
+    this.secretController = secretController;
+    this.aclDAO = aclDAOFactory.readonly();
+    this.clientDAO = clientDAOFactory.readwrite();
+  }
+
+  @VisibleForTesting
+  BatchSecretDeliveryResource(SecretController secretController, AclDAO aclDAO,
+                              ClientDAO clientDAO) {
+    this.secretController = secretController;
+    this.aclDAO = aclDAO;
+    this.clientDAO = clientDAO;
+  }
+
+  /**
+   * Retrieve Secret by name
+   *
+   * @param secrets the name of the Secrets to retrieve in batch
+   * @param client  the client performing the retrieval
+   * @return the secret with the specified name, if present and accessible to the client
+   * <p>
+   * responseMessage 200 Found and retrieved Secret with given name
+   * responseMessage 403 Secret is not assigned to Client
+   * responseMessage 404 Secret with given name not found
+   * responseMessage 500 Secret response could not be generated for given Secret
+   */
+  @Timed
+  @ExceptionMetered
+  @POST
+  @Consumes(APPLICATION_JSON)
+  public List<SecretDeliveryResponse> getBatchSecret(@Auth Client client, @Valid BatchSecretRequest request) {
+
+
+    List<SanitizedSecret> sanitizedSecrets = aclDAO.getBatchSanitizedSecretsFor(client, request.secrets);
+    List<Secret> secrets = secretController.getSecretsByName(request.secrets);
+
+    boolean clientExists = clientDAO.getClient(client.getName()).isPresent();
+
+
+    Iterator reqsecretsiter = request.secrets.iterator();
+
+
+    // The request fails whenever a single secret is requested that is not accessible
+    // The client is responsible for only requesting secrets they have permission for
+    while (reqsecretsiter.hasNext()) {
+      String secretname = (String) reqsecretsiter.next();
+
+      // We iterate over both arrays since the order of the two arrays might not match
+      SanitizedSecret sanitizedsecret = null;
```

Comment:
Since we're just checking whether these secrets exist, this section can be simplified to 
```
boolean secretExists = secrets.stream().anyMatch(s -> s.getName().equals(secretname));
boolean sanitizedSecretExists = sanitizedSecrets.stream().anyMatch(s -> s.getName.equals(secretname));
```
(although as noted above, I'd suggest renaming the variables)